src/components/game/UpdateGameForm.js

```diff
@@ -0,0 +1,118 @@
+                    <label htmlFor="number_of_players">Number of Players: </label>
+                    <input type="number" name="number_of_players" className="form-control"
+                        required autoFocus
+                        value={currentGame.number_of_players}
+                        onChange={changeGameState}
+                    />
+                </div>
+                <div className="form-group">
+                    <label htmlFor="skill_level">Skill Level: </label>
+                    <input type="number" name="skill_level" className="form-control"
+                        required autoFocus
+                        value={currentGame.skill_level}
+                        onChange={changeGameState}
+                    />
+                </div>
+                <div className="form-group">
+                    <label htmlFor="game_type">Game Type: </label>
+                    <select id="type" name="game_type" className="form-control"
+                        onChange={changeGameState}>
+                        <option value={0}>{currentGame.game_type?.label}</option>
+
+                        {
+                            gameTypes.map(gameType =>
+                                <option value={gameType.id} key={gameType.id}>{gameType.label}</option>
+                            )
+                        }
+
+
+                    </select>
+                </div>
+            </fieldset>
+
+            {/* TODO: create the rest of the input fields */}
+
+            <button type="submit"
+                onClick={evt => {
+                    // Prevent form from being submitted
+                    evt.preventDefault()
+
+                    if (currentGame.game_type.id) {
+                        const newGame = {
+                            maker: currentGame.maker,
+                            title: currentGame.title,
+                            numberOfPlayers: parseInt(currentGame.number_of_players),
+                            skillLevel: parseInt(currentGame.skill_level),
+                            gameType: parseInt(currentGame.game_type.id),
+                        }
+
+                        // Send POST request to your API
+                        updateGame(newGame, gameId)
+                            .then(() => navigate(`/games/${gameId}`))
+                    }
+                    else {
+                        const newGame = {
+                            maker: currentGame.maker,
+                            title: currentGame.title,
+                            numberOfPlayers: parseInt(currentGame.number_of_players),
+                            skillLevel: parseInt(currentGame.skill_level),
+                            gameType: parseInt(currentGame.game_type),
+                        }
+                        updateGame(newGame, gameId)
+                            .then(() => navigate(`/games/${gameId}`))
+                    }
+                }}
```

Comment:
Something is wrong here. The code is almost exactly the same in each conditional block, with the exception of `gameType` being set to `currentGame.game_type.id` in the first and set to `currentGame.game_type` in the second. If that is intentional, we at least need to DRY this up.